core/src/test/java/com/linecorp/armeria/server/AnnotatedHttpServiceTest.java

```diff
@@ -637,6 +688,77 @@ public void testParam() throws Exception {
         }
     }
 
+    @Test
+    public void testAdvancedAnnotatedHttpService() throws Exception {
```

Comment:
Added with the following comment. Need to think more about that.
```
            // No matched media type. It would be handled by the MyAnnotatedService8#sharedGet.
            // TODO(hyangtack) It is definitely wierd behavior because the request does not contain
            // 'accept: */*'. But we cannot check it strictly because we didn't have a media type
            // negotiation feature before. Currently we just assume that the request has a 'accept: */*'.
            httpGet = new HttpGet(rule.httpUri("/8/same/path"));
            httpGet.setHeader(org.apache.http.HttpHeaders.ACCEPT,
                              "application/json;charset=UTF-8;q=0.9, text/html;q=0.7");
            try (CloseableHttpResponse res = hc.execute(httpGet)) {
                assertThat(res.getStatusLine().toString(), is("HTTP/1.1 200 OK"));
                assertThat(EntityUtils.toString(res.getEntity()), is("GET"));
            }
```